src/software/ai/move_rule/move_rule_manager.cpp

```diff
@@ -0,0 +1,92 @@
+#include "software/ai/move_rule/move_rule_manager.h"
+
+#include "software/ai/hl/stp/tactic/all_tactics.h"
+
+
+std::set<MoveRule> MoveRuleManager::getMoveRules(const GameState &game_state,
+                                                 const Tactic &tactic)
+{
+    setCurrentMoveRulesFromGameState(game_state);
+    tactic.accept(*this);
+    return current_move_rules;
```

Comment:
In order to minimize side effects, could we make `setCurrentMoveRulesFromGameState` static and rename it so we can write this function like:
```
    std::set<MoveRule> move_rules = getCurrentMoveRulesFromGameState(game_state);
    tactic.accept(*this);
    move_rules.filter(whitelist); // I have no idea if this is how you do it, but populate a whitelist in the 
                                  // visitor and filter all values in whitelist out from move_rules
    return move_rules;

Reply:
hm the idea is to use current_move_rules as a global variable and both setCurrentMoveRulesFromGameState and accept would modify it. I'll update things to make that clear. (That's how the navigator works)